mkdocs.yml

```diff
@@ -98,7 +98,8 @@ nav:
   - ./internals/crates.md
   - ./internals/lifting_and_lowering.md
   - ./internals/object_references.md
-  - ./internals/async.md
```

Comment:
huh, so I guess I'm a little confused - you want to delete the existing async.md? It's not removed in this patch though, and that seems to cover things (including a cute mermaid diagram) that's not covered in the new docs.

Reply:
I'm not deleting, just renaming it to async overview to differentiate it from async FFI details.

Reply:
oops, sorry, I worked that out but must have left that stray comment!